components/sup/src/manager/service_updater/rolling_update_worker.rs

```diff
@@ -0,0 +1,265 @@
+use super::package_update_worker::PackageUpdateWorker;
+use crate::{census::{CensusGroup,
+                     CensusRing},
+            manager::service::{Service,
+                               Topology}};
+use habitat_butterfly;
+use habitat_common::owning_refs::RwLockReadGuardRef;
+use habitat_core::{package::PackageIdent,
+                   service::ServiceGroup};
+use parking_lot::RwLock;
+use std::{self,
+          sync::Arc,
+          time::Duration};
+use tokio::{self,
+            time};
+
+// The census ring does not have an async API. We make it look async by making API calls in a
+// loop with this delay after each call.
+const DELAY: Duration = Duration::from_secs(1);
+
+/// The role of the supervisor in the rolling update.
+enum Role {
+    Leader,
+    Follower,
+}
+
+/// While the follower is waiting for its turn, it can either be promoted to leader (the leader
+/// died) or it can be instructed to update to a specific ident.
+enum FollowerWaitForTurn {
+    PromotedToLeader,
+    // TODO (DM): This field should always be fully qualified. We need a
+    // type to encapsulate that.
+    UpdateTo(Option<PackageIdent>),
```

Comment:
Thanks for pointing this out. That option felt really weird to me as well, but I thought it would be painful to remove but turns out I am not sure why it was even an option. Probably, a vestige of old code. Addressed in [e8a6ea](https://github.com/habitat-sh/habitat/pull/7455/commits/e8a6ea4eb35c45c806534665fcb0b6911737f5cb)